From: Ben Hutchings <ben@decadent.org.uk>
Date: Mon, 27 Nov 2017 20:01:30 +0000
Subject: ip_fib: Avoid ABI change in 3.16.50
Forwarded: not-needed

Commit 3fb07daff8e9 "ipv4: add reference counting to metrics" changed
the type of fib_info::fib_metrics from u32 * to dst_metrics *.
However, dst_metrics begins with the same array-of-u32 as before, so
the change doesn't break existing readers; and all writers to the
field are built-in.  So hide the change from genksyms.

---
--- a/include/net/ip_fib.h
+++ b/include/net/ip_fib.h
@@ -110,11 +110,15 @@ struct fib_info {
 	unsigned char		fib_type;
 	__be32			fib_prefsrc;
 	u32			fib_priority;
+#ifndef __GENKSYMS__
 	struct dst_metrics	*fib_metrics;
 #define fib_mtu fib_metrics->metrics[RTAX_MTU-1]
 #define fib_window fib_metrics->metrics[RTAX_WINDOW-1]
 #define fib_rtt fib_metrics->metrics[RTAX_RTT-1]
 #define fib_advmss fib_metrics->metrics[RTAX_ADVMSS-1]
+#else
+	u32			*fib_metrics;
+#endif
 	int			fib_nhs;
 #ifdef CONFIG_IP_ROUTE_MULTIPATH
 	int			fib_power;
